three/three.cpp

```diff
@@ -170,6 +295,99 @@ PyObject *Three::_importFrom(const char *module, const char *name) {
     return NULL;
 }
 
+PyObject *Three::_findSubclassOf(PyObject *base, PyObject *module) {
+    if (base == NULL || !PyType_Check(base)) {
+        setError("base class is not of type 'Class'");
+        return NULL;
+    }
+
+    if (module == NULL || !PyModule_Check(module)) {
+        setError("module is not of type 'Module'");
+        return NULL;
+    }
+
+    PyObject *dir = PyObject_Dir(module);
+    if (dir == NULL) {
+        setError("there was an error calling dir() on module object");
+        return NULL;
+    }
+
+    PyObject *klass = NULL;
+    for (int i = 0; i < PyList_GET_SIZE(dir); i++) {
+        // get symbol name
+        char *symbol_name;
+        PyObject *symbol = PyList_GetItem(dir, i);
+        if (symbol != NULL) {
+            PyObject *bytes = PyUnicode_AsEncodedString(symbol, "UTF-8", "strict");
+
+            if (bytes != NULL) {
+                symbol_name = strdup(PyBytes_AsString(bytes));
+                Py_XDECREF(bytes);
+            } else {
+                continue;
+            }
+        } else {
+            // Gets exception reason
+            PyObject *reason = PyUnicodeDecodeError_GetReason(PyExc_IndexError);
+
+            // Clears exception and sets error
+            PyException_SetTraceback(PyExc_IndexError, Py_None);
+            setError(strdup(PyBytes_AsString(reason)));
+            goto done;
+        }
+
+        klass = PyObject_GetAttrString(module, symbol_name);
+        if (klass == NULL) {
+            continue;
+        }
+
+        // Not a class, ignore
+        if (!PyType_Check(klass)) {
+            Py_XDECREF(klass);
+            continue;
+        }
+
+        // Unrelated class, ignore
+        if (!PyType_IsSubtype((PyTypeObject *)klass, (PyTypeObject *)base)) {
+            Py_XDECREF(klass);
+            continue;
+        }
+
+        // `klass` is actually `base` itself, ignore
+        if (PyObject_RichCompareBool(klass, base, Py_EQ) == 1) {
```

Comment:
We should just check that it's `!= 0` since we don't want to continue on error (when the result is -1)